client/landing/gutenboarding/utils.ts

```diff
@@ -52,6 +55,7 @@ export function createSite( { siteTitle, siteUrl, theme, siteVertical }: CreateS
 		}
 
 		const siteSlug = urlToSlug( untrailingslashit( url ) );
+		onCreate();
```

Comment:
I expect this to be rather short lived (https://github.com/Automattic/wp-calypso/pull/39055 will replace it), but generally I'd prefer to return the promise as opposed to adding an `onCreate` parameter.

Fine to leave this here since it's temporary.